rpcwebsocket.go

```diff
@@ -1308,116 +1306,280 @@ out:
 			continue
 		}
 
-		// Requests with no ID (notifications) must not have a response per the
-		// JSON-RPC spec.
-		if request.ID == nil {
-			if !c.authenticated {
-				break out
-			}
-			continue
-		}
-
-		cmd := parseCmd(&request)
-		if cmd.err != nil {
-			if !c.authenticated {
+		// Determine if the request is single request or a batched request
+		switch request.(type) {
```

Comment:
Could you add a default case that does something like 
```go
default:
rpcslog.Debugf("unhandled request due to unknown type: %T", request)
```

Reply:
sure, on it.